runtime/ts/storage/in-memory-storage.ts

```diff
@@ -155,22 +148,39 @@ abstract class InMemoryStorageProvider extends StorageProviderBase {
     }
     return new InMemoryVariable(type, storageEngine, name, id, key);
   }
+
+  async ensureBackingStore() {
+    if (this.backingStore) {
+      return this.backingStore;
+    }
+    if (!this.pendingBackingStore) {
+      const key = this.storageEngine.baseStorageKey(this.backingType());
+      this.pendingBackingStore = this.storageEngine.baseStorageFor(this.type, key);
+      this.backingStore = await this.pendingBackingStore;
+    } else {
+      await this.pendingBackingStore;
+    }
+    return this.backingStore;
+  }
```

Comment:
Possible simplification:
```
async ensureBackingStore() {
  if (!this.backingStore) {
    if (!this.pendingBackingStore) {
      const key = this.storageEngine.baseStorageKey(this.backingType());
      this.pendingBackingStore = this.storageEngine.baseStorageFor(this.type, key);
    }
    this.backingStore = await this.pendingBackingStore;
    this.pendingBackingStore = null;
  }
  return [await?] this.backingStore;
}
```

Reply:
Looking at this again, it might not be correct to set pendingBackingStore to null after the await, since that could cause another invocation of ensureBackingStore awaiting at the same point to then write null to backingStore?

Reply:
Yeah, I definitely feel like I'm sinking here.